src/main/java/jude/task/Task.java

```diff
@@ -0,0 +1,110 @@
+package jude.task;
+
+/**
+ * The {@code Task} class models a task. A task has a description and an indicator depicting whether
+ * it has been done.
+ */
+public abstract class Task {
+    private final String description;
+    private boolean isDone;
+
+    /**
+     * Creates a new {@code Task} object with a given description and whether it has been done.
+     *
+     * @param description The description of the {@code Task}.
+     * @param isDone Whether the {@code Task} is marked as done.
+     */
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Returns whether the given {@code Task} object is marked as done.
+     *
+     * @return Whether the task is done.
+     */
+    public boolean getIsDone() {
+        return isDone;
+    }
+
+    /**
+     * Returns the description of the {@code Task} object.
+     *
+     * @return Description of the {@code Task} object.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Sets whether the {@code Task} is done.
+     *
+     * @param isDone Whether the {@code Task} is done.
+     */
+    public void setIsDone(boolean isDone) {
+        this.isDone = isDone;
+    }
+
+    /**
+     * Returns the status icon depicting whether the {@code Task} has been done.
+     *
+     * @return 'X' if {@code Task} is done, ' ' otherwise.
+     */
+    //@@author cheeheng-reused
+    //Reused from https://nus-cs2103-ay2223s1.github.io/website/schedule/week2/project.html
+    //with minor modifications
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Sets the status of the {@code Task} to 'done'.
+     */
+    public void markAsDone() {
```

Comment:
It seems like markAsDone(), markAsUndone() and setIsDone(isDone) all serve the purpose of setting this.isDone property. I'm not really sure about the value of setIsDone() being executed in other parts of your codebase as this is bad practice; Ideally it should only be called within the class itself. Perhaps you could consider just removing setIsDone() and using markAsDone() and markAsUndone() for the rest of the codebase. That way you can do proper error handling within the functions themselves too.